openvino_xai/methods/black_box/aise.py

```diff
@@ -151,74 +265,209 @@ def _preset_parameters(
+            target_indices = list(range(num_boxes))
 
-        loss = pred_score_preserve - pred_score_delete
+        self.num_iterations_per_kernel, self.divisors = self._preset_parameters(
+            preset,
+            num_iterations_per_kernel,
+            divisors,
+        )
 
-        self.pred_score_hist[self._current_kernel_width].append(pred_score_preserve - pred_score_delete)
+        self.solver_epsilon = solver_epsilon
+        self.locally_biased = locally_biased
 
-        loss *= -1  # Objective: minimize
-        return loss
+        self.input_size = infer_size_from_image(self.data_preprocessed)
+        original_size = infer_size_from_image(data)
+        self._mask_generator = GaussianPerturbationMask(self.input_size)
 
-    def _get_score(self, data_perturbed: np.array) -> float:
-        """Get model prediction score for perturbed input."""
-        x = self.model_forward(data_perturbed, preprocess=False)
-        x = self.postprocess_fn(x)
-        if np.max(x) > 1 or np.min(x) < 0:
-            x = sigmoid(x)
-        pred_scores = x.squeeze()  # type: ignore
-        return pred_scores[self.target]
+        saliency_maps = {}
+        for target in target_indices:
+            self.target_box = boxes[target]
+            self.target_label = labels[target]
 
-    def _kernel_density_estimation(self) -> np.ndarray:
-        """Aggregate the result per kernel with KDE."""
-        saliency_map_per_kernel = np.zeros((len(self.kernel_widths), self.input_size[0], self.input_size[1]))
-        for kernel_index, kernel_width in enumerate(self.kernel_widths):
-            kernel_masks_weighted = np.zeros(self.input_size)
-            for i in range(self.num_iterations_per_kernel):
-                kernel_params = self.kernel_params_hist[kernel_width][i]
-                kernel_mask = self._mask_generator.generate_kernel_mask(kernel_params)
-                score = self.pred_score_hist[kernel_width][i]
-                kernel_masks_weighted += kernel_mask * score
-            kernel_masks_weighted = kernel_masks_weighted / kernel_masks_weighted.max()
-            saliency_map_per_kernel[kernel_index] = kernel_masks_weighted
+            if self.target_box[0] >= self.target_box[2] or self.target_box[1] >= self.target_box[3]:
+                continue
 
-        saliency_map = saliency_map_per_kernel.sum(axis=0)
-        saliency_map /= saliency_map.max()
-        return saliency_map
+            self.kernel_params_hist = collections.defaultdict(list)
+            self.pred_score_hist = collections.defaultdict(list)
+
+            self._process_box()
+            saliency_map_per_target = self._run_synchronous_explanation()
+            if scale_output:
+                saliency_map_per_target = scaling(saliency_map_per_target)
+            saliency_maps[target] = saliency_map_per_target
+
+            self._update_metadata(boxes, scores, labels, target, original_size)
```

Comment:
I feel a bit awkward here. `Method` instance can process multiple series inputs by definition but we have some metadata for previous input saved as internal state. Can we feed the bbox info to the visualizer via other channels?

Reply:
This is a good point. 
Method's `generate_saliency_map` interface should take care of instance state cleaning (the same way as it updates some instance attributes, e.g. `self.input_size`).

Reply:
> Can we feed the bbox info to the visualizer via other channels?

Definitely, we can. Obvious option is to return this metadata with a saliency map. But I propose not to overload `generate_saliency_map` interface, let it just return saliency maps and nothing else.

So, the other option is to save some info that we need in a method instance state, e.g. `method.metadata`, taking care of cleaning. 
What so you think?

Reply:
Understood. Let's keep it as is until we learn more use cases and find better interface.